apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/manager/room/RoomManager.kt

```diff
@@ -126,18 +123,28 @@ class RoomManager @Inject constructor(
     }
 
     override fun onDisconnected(room: Room, twilioException: TwilioException?) {
-        Timber.i("onDisconnected -> room sid: %s", room.sid)
+        Timber.i("onDisconnected -> room sid: ${room.sid}")
+        Timber.i("onDisconnected -> room exception message: ${twilioException?.message}")
+        Timber.i("onDisconnected -> room exception code: ${twilioException?.code}")
+        Timber.i("onDisconnected -> Lifecycle state: ${lifecycleOwner?.lifecycle?.currentState}")
+
+        // This condition is most when the room is disconnected by the local participant. In example, clicking the UI to disconnect the room.
+        if (this.room == null) return
+
         twilioException?.code?.apply {
-            if (this == TwilioException.ROOM_ROOM_COMPLETED_EXCEPTION) {
-                disconnect(RoomDisconnectionType.StreamEndedByHost)
+            when (this) {
+                TwilioException.ROOM_ROOM_COMPLETED_EXCEPTION -> disconnect(RoomDisconnectionType.StreamEndedByHost)
+                TwilioException.PARTICIPANT_NOT_FOUND_EXCEPTION -> disconnect(RoomDisconnectionType.SpeakerMovedToViewersByHost)
             }
+        } ?: run {
+            disconnect(RoomDisconnectionType.SpeakerMovedToViewersByHost)
```

Comment:
What happens when there is some random error that we don't specifically handle? Don't we need to clean up and pass that error to the UI to display? I don't see how this code handles this case but maybe I don't understand the Kotlin.

Reply:
Yeah, this is cleanin up. I if you go to the `disconnect` method implementation (Line 72)... you will find it, internally it's cleaning up:
```
private fun disconnect(disconnectionType: RoomDisconnectionType?, notifyStateEvent: Boolean = true) {
        cleanUp()
        if (notifyStateEvent)
            _onStateEvent.value = RoomViewEvent.OnDisconnected(disconnectionType)
 }
```

Reply:
How does this error handling cover other errors like `TVIErrorParticipantDuplicateIdentityError = 53205`?

Errors that do not require specific handling should still be handled in a generic way. Can you explain the path for this case?

For example this is how iOS handles other errors: https://github.com/twilio/twilio-live-interactive-video/blob/fea431032bffde35f5fb93b6ff8afb2caf95c3de/apps/ios/LiveVideo/LiveVideo/Twilio/Room/RoomManager.swift#L91

Reply:
done!